tests/test-schema.js

```diff
@@ -0,0 +1,142 @@
+    }
+
+    return errors;
+}
+
+
+describe('All data', function() {
+  describe('in conditions.js', function() {
+    it('should validate against conditions schema', function() {
+      var errors = validateData('Conditions', Data.conditions, Schema.conditions);
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
+    });
+  });
+
+  describe('in damage-deck-core.js', function() {
+    it('should validate against damage-deck schema', function() {
+      var errors = validateData('DamageDeckCore', Data.damageDeckCore, Schema.damageDeck);
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
+    });
+  });
+
+  describe('in damage-deck-core-tfa.js', function() {
+    it('should validate against damage-deck schema', function() {
+      var errors = validateData('DamageDeckCore', Data.damageDeckCoreTfa, Schema.damageDeck);
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
+    });
+  });
+
+  describe('in pilots.js', function() {
+    it('should validate against pilots schema', function() {
+      var errors = validateData('Pilot', Data.pilots, Schema.pilots);
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
+    });
+  });
+
+  describe('in ships.js', function() {
+    it('should validate against ships schema or huge-ships schema', function() {
+      var errors = [];
+      var errorText, header, schema;
+
+      for (var i = 0; i < Data.ships.length; i++) {
+
+        if (Data.ships[i].size == 'huge') {
+          header = 'Huge Ship';
+          schema = Schema.hugeShips;
+        } else if (['large', 'small'].indexOf(Data.ships[i].size) != -1) {
+          header = 'Ship';
+          schema = Schema.ships;
+        } else {
+          errorText = '\t<Ship index=' + i.toString() + '>';
+          errorText += ':\n\t\t- should have required property \'size\'\n';
+          errors.push(errorText);
+          continue;
+        }
+
+        if (!schema(Data.ships[i])) {
+
+          if (Data.ships[i].name) {
+            errorText = '\t<' + header + ' name="' + Data.ships[i].name + '">:\n\t\t';
+          } else {
+            errorText = '\t<' + header + ' index=' + i.toString() + '>:\n\t\t';
+          }
+
+          for (var j = 0; j < schema.errors.length; j++) {
+            errorText += '- ' + schema.errors[j].dataPath + ' ' + schema.errors[j].message + '\n\t\t';
+          }
+
+          errors.push(errorText);
+        }
+      }
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
+    });
+  });
+
+  describe('in sources.js', function() {
+    it('should validate against sources schema', function() {
+      var errors = validateData('Source', Data.sources, Schema.sources);
+
+      if (errors.length > 0) {
+        assert.fail(false, true, '\n' + errors.join('\n'));
+      }
```

Comment:
This is repeated for every test which makes the file a lot longer than it has to be. Would you be against moving the assertion to `validateData()` instead of returning an array? Each test could be as simple as:
```js
it('should validate against sources schema', function() {
  validateData('Source', Data.sources, Schema.sources);
});
```